packages/frontend/src/pages/Send/Send.tsx

```diff
@@ -287,14 +287,34 @@ const Send: FC = () => {
   const needsTokenForFee = useNeedsTokenForFee(fromNetwork)
 
   useEffect(() => {
-    if (needsTokenForFee && fromNetwork) {
-      setNeedsNativeTokenWarning(
-        `Add ${fromNetwork.nativeTokenSymbol} to your account on ${fromNetwork.name} for the transaction fee.`
-      )
-    } else {
-      setNeedsNativeTokenWarning('')
+    const update = async () => {
+      let warning = ''
+      try {
+        if (needsTokenForFee && fromNetwork && toNetwork && fromTokenAmountBN && fromBalance && deadline) {
+          const options = {
+            balance: fromBalance,
+            token: sourceToken,
+            fromNetwork,
+            toNetwork,
+            deadline,
+          }
+    
+          var estimatedGasCost = await estimateSend(options) 
+
+          if (estimatedGasCost && fromBalance?.lt(estimatedGasCost)) {
```

Comment:
if this block of code should **only** run if the user is trying to send a native token asset, let's explicitly state it in the `if` condition -- `sourceToken.isNativeToken`.

Reply:
matter of fact, we can be just as explicit while exiting this hook quicker if we check `sourceToken.isNativeToken` in the earlier `if` condition